packages/devtools_app/lib/src/screens/memory/panes/chart/primitives.dart

```diff
@@ -0,0 +1,68 @@
+// Copyright 2022 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+enum ChartType {
+  DartHeaps,
+  AndroidHeaps,
+}
+
+/// Automatic pruning of collected memory statistics (plotted) full data is
+/// still retained. Default is the best view each tick is 10 pixels, the
+/// width of an event symbol e.g., snapshot, monitor, etc.
+enum ChartInterval {
```

Comment:
make this an enhanced enum so that we don't have to have methods `chartDuration` and `displayDuration`. See this recent PR for an example of this: https://github.com/flutter/devtools/pull/4803/files#diff-6f087d6cdc6c21288a3321f6fbf556393aee9f2bcf167ab040d21fd44887d4c3